openpype/hosts/traypublisher/csv_publish.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+import pyblish.api
+import pyblish.util
+
+from openpype.client import get_asset_by_name
+from openpype.lib.attribute_definitions import FileDefItem
+from openpype.pipeline import install_host
+from openpype.pipeline.create import CreateContext
+
+from openpype.hosts.traypublisher.api import TrayPublisherHost
+
+
+def csvpublish(
+    csv_filepath,
+    project_name,
+    asset_name,
+    task_name=None,
+    username=None,
+    targets=None
+):
+    """Publish CSV file.
+
+    Args:
+        csv_filepath (str): Path to CSV file.
+        project_name (str): Project name.
+        asset_name (str): Asset name.
+        task_name (Optional[str]): Task name.
+        username (Optional[str]): User name.
+        hostname (Optional[str]): Host name.
+        targets (Optional[list[str]]): List of targets.
+    """
+
+    # initialization of host
+    host = TrayPublisherHost()
+    install_host(host)
+
+    # setting host context into project
+    host.set_project_name(project_name)
+
+    # add asset context to environment
+    # TODO: perhaps this can be done in a better way?
+    os.environ.update({
+        "AVALON_PROJECT": project_name,
+        "AVALON_ASSET": asset_name,
+        "AVALON_TASK": task_name
```

Comment:
Arguments are suggesting that `task_name` can be `None`, in that case would this crash, because `os.environ` does allow only `str` type.

Reply:
applied via 9c1e679b59bf56728c81a3b60aef792215fe61da